[ci] only upload summarized test stats for PRs

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #80295

We are uploading too many tests which is stressing out Rockset. We do
not really use granular per-test statistics for PRs, so change to
uploading summary information for PRs.

We will still be uploading per-test stats from master.

suo committed:
[ci] only upload summarized test stats for PRs

We are uploading too many tests which is stressing out Rockset. We do
not really use granular per-test statistics for PRs, so change to
uploading summary information for PRs.

We will still be uploading per-test stats from master.

[ghstack-poisoned]

suo commented:
@pytorchbot merge -f

janeyx99 commented on tools/stats/upload_test_stats.py:
```diff
@@ -162,7 +197,18 @@ def get_tests(
+    # volume of writes we do to Rockset.
+    upload_to_rockset("test_run_summary", summarize_test_cases(test_cases))
+
+    if args.head_branch != "master":
```
wait why is this a !=

suo replied:
yeah this is wrong lol

suo commented:
@pytorchbot revert -m "reverting and relanding with a small bugfix" -c nosignal

suo committed:
Update on "[ci] only upload summarized test stats for PRs"


We are uploading too many tests which is stressing out Rockset. We do
not really use granular per-test statistics for PRs, so change to
uploading summary information for PRs.

We will still be uploading per-test stats from master.

[ghstack-poisoned]

janeyx99 reviewed (approved):
This PR also just gets rid of test suite aggregation and opts for the manual aggregation for all cases, yes?

suo commented:
> This PR also just gets rid of test suite aggregation and opts for the manual aggregation for all cases, yes?

yes, sadly xmlrunner does not produce good output for `test-suite` so that aggregation end up not being useful

suo commented:
@pytorchbot merge -f
